AddonExample/AddonExampleEventSource.cs

```diff
@@ -0,0 +1,177 @@
+        {
+            Config = AddonExampleEventSourceConfig.LoadConfig(config);
+        }
+
+        public override void SaveConfig(IPluginConfig config)
+        {
+            Config.SaveConfig(config);
+        }
+
+        public override void Start()
+        {
+            // Reset Event Dispatcher
+            OnAddonExampleOriginalTimerFired -= (e) => DispatchToJS(e);
+            OnAddonExampleOriginalTimerFired += (e) => DispatchToJS(e);
+            OnAddonExampleEmbeddedTimerFired -= (e) => DispatchToJS(e);
+            OnAddonExampleEmbeddedTimerFired += (e) => DispatchToJS(e);
+
+            // Start the embedded timer when using it.
+            // Call base.Start() or timer.Change(0, interval) to start the embedded timer manually.
+            base.Start();
+            //timer.Change(0, 1000);
+
+            // Start Original Timer
+            originalTimer = new System.Timers.Timer()
+            {
+                Interval = 1000,
+                AutoReset = true,
+            };
+            originalTimer.Elapsed += (obj, args) =>
+            {
+                OnAddonExampleOriginalTimerFired(new JSEvents.OriginalTimerFiredEvent(Config.ExampleString + " fired!"));
+            };
+            originalTimer.Start();
+        }
+
+        public override void Stop()
+        {
+            // Stop original timer
+            originalTimer.Stop();
+
+            // Stop the embedded timer when using it.
+            // Call base.stop() or timer.Change(-1, -1) to stop the embedded timer manually.
+            base.Stop();
+            //timer.Change(-1, -1);
+        }
+
+        /// <summary>
+        /// This method is called periodically when using the embedded timer.
+        /// </summary>
+        protected override void Update()
+        {
+            OnAddonExampleEmbeddedTimerFired(new JSEvents.EmbeddedTimerFiredEvent("fired!"));
+        }
+
+        public override void Dispose()
+        {
+            base.Dispose();
+        }
+
+        // Sends an event called |event_name| to javascript, with an event.detail that contains
+        // the fields and values of the |detail| structure.
+        public void DispatchToJS(JSEvent e)
+        {
+            JObject ev = new JObject();
+            ev["type"] = e.EventName();
+            ev["detail"] = JObject.FromObject(e);
+            DispatchEvent(ev);
+        }
+
+        public void LogDebug(string format, params object[] args)
+        {
+            this.Log(LogLevel.Debug, format, args);
+        }
+
+        public void LogError(string format, params object[] args)
+        {
+            this.Log(LogLevel.Error, format, args);
+        }
+
+        public void LogInfo(string format, params object[] args)
+        {
+            this.Log(LogLevel.Warning, format, args);
+        }
+
+        public void LogWarning(string format, params object[] args)
+        {
+            this.Log(LogLevel.Info, format, args);
+        }
```

Comment:
Why did you add the `Log*` methods?
Calling `this.LogDebug("Some stuff with {0} value.", "a");` doesn't seem any simpler than `this.Log(LogLevel.Debug, "Some stuff with {0} value.", "a");`.